politeiad/plugins/comments/comments.go

```diff
@@ -362,7 +393,7 @@ type NewReply struct {
 // PublicKey is the user's public key that is used to verify the signature.
 //
 // Signature is the user signature of the:
-// State + Token + ParentID + Comment + ExtraData + ExtraDataHint
+// State + Token + ParentID + CommentID + Comment + ExtraData + ExtraDataHint
```

Comment:
The comment for the CommentAdd record also needs to be updated.

```
// CommentAdd is the structure that is saved to disk when a comment is created
// or edited.
//
// PublicKey is the user's public key that is used to verify the signature.
//
// Signature is the user signature of the:
// State + Token + ParentID + Comment + ExtraData + ExtraDataHint
//
// Receipt is the server signature of the user signature.
//
// The PublicKey, Signature, and Receipt are all hex encoded and use the
// ed25519 signature scheme.
type CommentAdd struct {
	// Data generated by client
	UserID    string       `json:"userid"`    // Unique user ID
	State     RecordStateT `json:"state"`     // Record state
	Token     string       `json:"token"`     // Record token
	ParentID  uint32       `json:"parentid"`  // Parent comment ID
	Comment   string       `json:"comment"`   // Comment
	PublicKey string       `json:"publickey"` // Pubkey used for Signature
	Signature string       `json:"signature"` // Client signature

	// Metadata generated by server
	CommentID uint32 `json:"commentid"` // Comment ID
	Version   uint32 `json:"version"`   // Comment version
	Timestamp int64  `json:"timestamp"` // Received UNIX timestamp
	Receipt   string `json:"receipt"`   // Server signature of client signature

	// Optional fields to be used freely
	ExtraData     string `json:"extradata,omitempty"`
	ExtraDataHint string `json:"extradatahint,omitempty"`
}
```

Reply:
just noticed that the signature structure depends whether the CommentAdd is associated with
a comment create or comment edit.

Reply:
came up with:

```
// CommentAdd is the structure that is saved to disk when a comment is created
// or edited.
//
// PublicKey is the user's public key that is used to verify the signature.
//
// The structure of the signature field depends on whether the CommentAdd is
// associated with a new comment or a comment edit:
//
//   1. When a comment is created it's the user signature of the:
//   State + Token + ParentID + Comment + ExtraData + ExtraDataHint.
//
//   2. When a comment is edited it's the user signature of the:
//   State + Token + ParentID + CommentID + Comment + ExtraData + ExtraDataHint.
//
// Receipt is the server signature of the user signature.
//
// The PublicKey, Signature, and Receipt are all hex encoded and use the
// ed25519 signature scheme.
```

Reply:
Looks good.